### PR TITLE
Added support for accessing execution location through script

### PR DESCRIPTION
**Pull request by vedkribhu**

### CODE REVIEW

File: lib/runner/extensions/event.command.js [hunk 1]
```diff
@@ -92,6 +92,26 @@ sanitizeFiles = function (request, callback) {
...
+
+    while (current) {
+        if (current.name) {
+            path = current.name + (path !== '' ? '/' : '') + path;
+        }
+
+        current = current.__parent;
+    }
+
+    return path;
```
Comment (coditva) [inline lib/runner/extensions/event.command.js:112]: We can avoid the complex logic here:

```suggestion
function getItemPath (item) {
    const pathParts = [];
    let current = item;

    while (current) {
        if (current.name) {
            pathParts.push(current.name);
        }

        current = current.__parent;
    }

    return pathParts.reverse().join('/');
```

File: lib/runner/extensions/event.command.js [hunk 2]
```diff
@@ -92,6 +92,26 @@ sanitizeFiles = function (request, callback) {
     return callback(null, request, sanitisedFiles);
 };
 
+/**
+ * Return the item path
+ *
+ * @param {any} item - sdk item
```
Comment (coditva) [inline lib/runner/extensions/event.command.js:98]: Why `any`?

File: test/integration/sandbox-libraries/pm.test.js [hunk 3]
```diff
@@ -578,4 +578,322 @@ describe('sandbox library - pm api', function () {
             });
         });
     });
+
+    describe('pm.info properties', function () {
+        describe('pm.info.requestPath', function () {
+            it('should have path as empty array for un-saved requests', function (done) {
```
Comment (coditva) [inline test/integration/sandbox-libraries/pm.test.js:584]: What does "un-saved" mean for `postman-runtime`? It's a Postman App concept leaking here.

File: test/integration/sandbox-libraries/pm.test.js [hunk 4]
```diff
@@ -578,4 +578,322 @@ describe('sandbox library - pm api', function () {
...
+                this.run({
+                    collection: {
+                        item: [{
+                            event: [
+                                {
+                                    listen: 'prerequest',
```
Comment (coditva) [inline test/integration/sandbox-libraries/pm.test.js:590]: What about tests script?

File: test/integration/sandbox-libraries/pm.test.js [hunk 5]
```diff
@@ -578,4 +578,322 @@ describe('sandbox library - pm api', function () {
...
+                    const collPreConsole = testrun.console.getCall(0).args.slice(2);
+
+                    expect(collPreConsole).to.deep.include.ordered.members(['path',
+                        []]);
+                    done();
+                });
+            });
+
+
+            describe('saved request cases', function () {
```
Comment (coditva) [inline test/integration/sandbox-libraries/pm.test.js:642]: The suite name should be a noun or an adjective. `describe` with `it` should form a sentence.
e.g.
```javascript
describe('An apple', () => {
  describe('with a worm', () => {
    it('should be thrown away', () => {});
  });
});

// forms a sentence: An apple, with a worm, should be thrown away.
```

File: test/integration/sandbox-libraries/pm.test.js [hunk 6]
```diff
@@ -578,4 +578,322 @@ describe('sandbox library - pm api', function () {
...
+                        }
+                    }, function (_err, results) {
+                        testrun = results;
+                        expect(testrun).to.be.ok;
+                        done();
+                    });
+                });
+
+                it(`should have path as correct path for a request nested inside folder when printed from
+                pre-request script of collection`, function () {
```
Comment (coditva) [inline test/integration/sandbox-libraries/pm.test.js:802]: What is the convention in this repo? Should test names contain a newline?
```suggestion
                it(
                  'should have path as correct path for a request nested inside folder when printed from pre-request script of collection',
                  function () {
```

File: test/integration/sandbox-libraries/pm.test.js [hunk 7]
```diff
@@ -578,4 +578,322 @@ describe('sandbox library - pm api', function () {
...
+                    const collPreConsole = testrun.console.getCall(5).args.slice(2);
+
+                    expect(collPreConsole).to.deep.include.ordered.members([
+                        'request pre',
+                        ['collection-0', 'request-0']
+                    ]);
+                });
+
+                it(`should have path as correct path for a request inside collection when printed from
+                test-request script of collection`, function () {
```
Comment (coditva) [inline test/integration/sandbox-libraries/pm.test.js:874]:
```suggestion
                tests script of collection`, function () {
```

File: test/integration/sandbox-libraries/pm.test.js (hunk 6 above)
Comment (vedkribhu) [reply test/integration/sandbox-libraries/pm.test.js:861]: @codenirvana can guide here, not able to find more such instances.

**Comment by coditva** [comment]
Need to update the `CHANGELOG.yaml`.

**Review by coditva — COMMENTED** [review]
Looks good. Can merge once `postman-sandbox` is bumped.

File: lib/runner/extensions/event.command.js [hunk 8]
```diff
@@ -457,6 +457,8 @@ module.exports = {
                     shouldSkipExecution = true;
                 });
 
+                const currentEventItem = event.parent && event.parent();
```
Comment (coditva) [inline lib/runner/extensions/event.command.js:460]: Can we add comment about why the current item is `.parent`?

**Review by appurva21 — APPROVED** [review]

**Review by codenirvana — APPROVED** [review]

**Pull request merged**